### ISSUE TITLE
different results from testing the same image twice

### ISSUE THREAD

**Issue by huihui6666** [mbortolon97/6dgs#11]
Hi, I train the the dataset Mid-360 and I test the same image twice,but the reason is very different. the reasan is below ,could you tell me where the problem might be?
`(gaussian_splatting_env) lihui@lihui-tower:~/Documents/6dgs$ python test.py --exp_dir_filepath output/mip_360_bicycle_d9921656-d --image_path /home/lihui/Downloads/360_v2/bicycle/images_2/_DSC8679.JPG
Using cache found in /home/lihui/.cache/torch/hub/facebookresearch_dinov2_main
/home/lihui/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/swiglu_ffn.py:51: UserWarning: xFormers is not available (SwiGLU)
  warnings.warn("xFormers is not available (SwiGLU)")
/home/lihui/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/attention.py:33: UserWarning: xFormers is not available (Attention)
  warnings.warn("xFormers is not available (Attention)")
/home/lihui/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/block.py:40: UserWarning: xFormers is not available (Block)
  warnings.warn("xFormers is not available (Block)")
Checkpoint already exist, skip training phase
/home/lihui/Documents/6dgs/test.py:38: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  ckpt_dict = torch.load(id_module_ckpt_path, map_location=device)
Loading complete starting the test...
0it [00:00, ?it/s]/home/lihui/Documents/6dgs/pose_estimation/line_intersection.py:237: UserWarning: Using torch.cross without specifying the dim arg is deprecated.
Please either pass the dim explicitly or simply use torch.linalg.cross.
The default value of dim will change to agree with that of linalg.cross in a future release. (Triggered internally at ../aten/src/ATen/native/Cross.cpp:62.)
  xaxis = torch.cross(up, direction)
1it [00:00,  6.57it/s]
Average loss score:  -1.0
Average Recall:  -1.0
Time per element:  0.15367388725280762
Translation Error:  4.055395603179932
Angular Error:  49.129180908203125
Smallest translation error:  0
[{'sequence_id': '', 'category_name': '', 'frame_id': 0, 'loss': 0.009999999776482582, 'scores_loss': -1.0, 'recall': -1.0, 'total_optimization_time_in_ms': 0.0, 'pred_c2w': [[0.6550666689872742, -0.12595845758914948, 0.7449980974197388, -3.1940786838531494], [0.15672661364078522, 0.9872130155563354, 0.029102729633450508, 1.1860064268112183], [-0.739137589931488, 0.09769678860902786, 0.6664314866065979, -2.1994283199310303], [0.0, 0.0, 0.0, 1.0]], 'gt_c2w': [[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 1.0, 0.0], [0.0, 0.0, 0.0, 1.0]]}]
(gaussian_splatting_env) lihui@lihui-tower:~/Documents/6dgs$ python test.py --exp_dir_filepath output/mip_360_bicycle_d9921656-d --image_path /home/lihui/Downloads/360_v2/bicycle/images_2/_DSC8679.JPG
Using cache found in /home/lihui/.cache/torch/hub/facebookresearch_dinov2_main
/home/lihui/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/swiglu_ffn.py:51: UserWarning: xFormers is not available (SwiGLU)
  warnings.warn("xFormers is not available (SwiGLU)")
/home/lihui/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/attention.py:33: UserWarning: xFormers is not available (Attention)
  warnings.warn("xFormers is not available (Attention)")
/home/lihui/.cache/torch/hub/facebookresearch_dinov2_main/dinov2/layers/block.py:40: UserWarning: xFormers is not available (Block)
  warnings.warn("xFormers is not available (Block)")
Checkpoint already exist, skip training phase
/home/lihui/Documents/6dgs/test.py:38: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  ckpt_dict = torch.load(id_module_ckpt_path, map_location=device)
Loading complete starting the test...
0it [00:00, ?it/s]/home/lihui/Documents/6dgs/pose_estimation/line_intersection.py:237: UserWarning: Using torch.cross without specifying the dim arg is deprecated.
Please either pass the dim explicitly or simply use torch.linalg.cross.
The default value of dim will change to agree with that of linalg.cross in a future release. (Triggered internally at ../aten/src/ATen/native/Cross.cpp:62.)
  xaxis = torch.cross(up, direction)
1it [00:00,  6.40it/s]
Average loss score:  -1.0
Average Recall:  -1.0
Time per element:  0.15771746635437012
Translation Error:  2.853109359741211
Angular Error:  69.36100006103516
Smallest translation error:  0
[{'sequence_id': '', 'category_name': '', 'frame_id': 0, 'loss': 0.009999999776482582, 'scores_loss': -1.0, 'recall': -1.0, 'total_optimization_time_in_ms': 0.0, 'pred_c2w': [[0.36006733775138855, -0.06667597591876984, 0.9305405616760254, -2.2431397438049316], [0.20455999672412872, 0.9788124561309814, -0.009018530137836933, 1.0056270360946655], [-0.9102234840393066, 0.193598672747612, 0.3660776615142822, -1.4481958150863647], [0.0, 0.0, 0.0, 1.0]], 'gt_c2w': [[1.0, 0.0, 0.0, 0.0], [0.0, 1.0, 0.0, 0.0], [0.0, 0.0, 1.0, 0.0], [0.0, 0.0, 0.0, 1.0]]}]`
